### PR TITLE
Add X-Ray profiling for daemons and subsegments for GET /v1/files

### PR DESCRIPTION
**Pull request by Bento007**
## Need (User story)
As a DSS operator
I want to profile uninstrumented (unpatched) DSS application logic (code not in `boto3` or `requests`)
so that I know what parts of an endpoint's logic are slow.

## Approach
Add xray tracing subsegments to the GET /files logic.

Adding xray tracing subsegments to the GET /files logic will allow us to profile unpatched code.

## Deployment Instructions

Must be deployed from a privileged account in order to update AWS Permissions for lambdas

### CODE REVIEW

File: dss/xraytrace.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import typing
+
+DSS_XRAY_TRACE = int(os.environ.get('DSS_XRAY_TRACE', '0')) > 0  # noqa
+
+
+def config_xray() -> None:
+    if DSS_XRAY_TRACE:  # noqa
+        from aws_xray_sdk.core import xray_recorder, patch_all
+        patch_all()
+        xray_recorder.configure(context_missing='LOG_ERROR')
```
Comment (mweiden) [inline dss/xraytrace.py]: Do you need to set the context here?

File: daemons/dss-index/app.py [hunk 2]
```diff
@@ -16,7 +19,10 @@
 from dss.logging import configure_lambda_logging
 from dss.util.time import RemainingLambdaContextTime, AdjustedRemainingTime
 
+
 app = domovoi.Domovoi(configure_logs=False)
+
+
```
Comment (mweiden) [inline daemons/dss-index/app.py]: nit: added whitespace?

**Comment by mweiden** [comment]
Also you've got linter issues in Travis

File: dss/xraytrace.py (hunk 1 above)
Comment (Bento007) [reply dss/xraytrace.py]: I thought we only needed that when we weren't using the lambda_context within a lambda.

**Pull request merged**